app/user_cleanup.go

```diff
@@ -0,0 +1,137 @@
+package app
+
+import (
+	"strings"
+	"time"
+
+	"github.com/rancher/rancher/pkg/systemaccount"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+type userCleanup struct {
+	users         v3.UserInterface
+	userLister    v3.UserLister
+	clusterLister v3.ClusterLister
+	projectLister v3.ProjectLister
+}
+
+func cleanupOrphanedSystemUsers(management *config.ManagementContext) {
+	u := userCleanup{
+		users:         management.Management.Users(""),
+		userLister:    management.Management.Users("").Controller().Lister(),
+		clusterLister: management.Management.Clusters("").Controller().Lister(),
+		projectLister: management.Management.Projects("").Controller().Lister(),
+	}
+
+	go wait.PollImmediate(time.Hour*24, 0, func() (bool, error) {
```

Comment:
we have to pass the context to the go routine - the examples can be found in the crone library